### PR TITLE
Use popover dismiss instead of emitter on unpin job (#2dewqjn)

### PR DESCRIPTION
**Pull request by rathoreprashant**
### Related Issues
usepopover.onDidDismiss on unpin jobs

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/components/JobActionsPopover.vue [hunk 1]
```diff
@@ -54,8 +53,11 @@ export default defineComponent({
     })
   },
   methods: {
-    closePopover() {
-      popoverController.dismiss({ dismissed: true });
+    closePopover(enumId?: any) {
+      popoverController.dismiss({
+        dismissed: true,
```
Comment (ymaheshwari1) [inline src/components/JobActionsPopover.vue:58]: I think there is no need to pass `dismissed: true`

File: src/views/Pipeline.vue [hunk 2]
```diff
@@ -502,6 +502,11 @@ export default defineComponent({
         event: ev,
         componentProps: { job }
       });
+
+      popover.onDidDismiss()
+        .then((result) => {
+        if (!this.getPinnedJobs.includes(result.data.systemJobEnumId)) this.updateSelectedPinnedJob(result.data.systemJobEnumId);
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:508]:
```suggestion
        if (!this.getPinnedJobs.includes(result.data.systemJobEnumId)) {
          this.updateSelectedPinnedJob(result.data.systemJobEnumId);
        }
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/JobActionsPopover.vue (hunk 1 above)
Comment (rathoreprashant) [reply src/components/JobActionsPopover.vue]: Removed sir

File: src/views/Pipeline.vue (hunk 2 above)
Comment (rathoreprashant) [reply src/views/Pipeline.vue]: Improved sir

File: src/views/Pipeline.vue [hunk 3]
```diff
@@ -570,13 +576,6 @@ export default defineComponent({
     // state does not gets updated and hence the job configuration component takes it space in DOM
     await this.store.dispatch('job/updateCurrentJob', { job: {} });
   },
-  mounted(){
-    emitter.on("pinnedJobsUpdated", (this as any).updateSelectedPinnedJob);
-  },
-  unmounted(){
-    emitter.off('jobUpdated', this.updateJobs);
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:577]: We don't need to remove this jobUpdated emitter.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 3 above)
Comment (rathoreprashant) [reply src/views/Pipeline.vue:577]: ok sir I'll fix this

Comment (rathoreprashant) [reply src/views/Pipeline.vue:577]: Fixed sir
please have a look

File: src/views/Pipeline.vue [hunk 4]
```diff
@@ -502,7 +502,14 @@ export default defineComponent({
...
       });
-      return popover.present()
+
+      popover.onDidDismiss().then((result) => {
+        const jobEnumId = result.data.systemJobEnumId;
+        if (jobEnumId && !this.getPinnedJobs.includes(jobEnumId)){
+          this.updateSelectedPinnedJob(jobEnumId);
+        }
+      });
+        return popover.present()
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:512]: Improve indentation.

File: src/views/Pipeline.vue [hunk 5]
```diff
@@ -502,7 +502,14 @@ export default defineComponent({
         event: ev,
         componentProps: { job }
       });
-      return popover.present()
+
+      popover.onDidDismiss().then((result) => {
+        const jobEnumId = result.data.systemJobEnumId;
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:507]: Add a condition that whether we have data in the result or not.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 4 above)
Comment (rathoreprashant) [reply src/views/Pipeline.vue]: fixed

File: src/views/Pipeline.vue (hunk 5 above)
Comment (rathoreprashant) [reply src/views/Pipeline.vue]: updated sir

**Review by ymaheshwari1 — APPROVED** [review]

File: src/views/Pipeline.vue [hunk 6]
```diff
@@ -502,6 +502,13 @@ export default defineComponent({
         event: ev,
         componentProps: { job }
       });
+
+      popover.onDidDismiss().then((result) => {
+        const jobEnumId = result.data?.systemJobEnumId;
+        if (jobEnumId && !this.getPinnedJobs.includes(jobEnumId)){
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:508]: Why this is needed?

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 6 above)
Comment (rathoreprashant) [reply src/views/Pipeline.vue:508]: when a job is pinned and selected
Now if the user unpin that job, we have to updateSelectedPinnedJob without updating, the job is unpinned but that job is still filtered

**Comment by adityasharma7** [comment]
Closing as this will not be needed for now. We will be using emitter only we are doing pin/unpin action from different components

**Pull request closed without merging**